File: javaRestAssured/src/test/java/io/swagger/petstore/PetTestsInObjectStyle.java
package io.swagger.petstore;

import io.swagger.petstore.controllers.JPetController;
import io.swagger.petstore.models.JMessage;
import io.swagger.petstore.models.JPet;
import org.apache.commons.lang3.RandomStringUtils;
import org.junit.Assert;
import org.junit.Test;

public class PetTestsInObjectStyle {

    @Test
    public void addNewPetToStoreTest() {
        JPet testPet = new JPet(null,
                "Pet_" + RandomStringUtils.randomAlphabetic(8), //Pet name
                RandomStringUtils.randomNumeric(8), //Pet ID
                null, null,
                "available"); //Pet status

        JPet petResponse = new JPetController().addNewPet(testPet);
        Assert.assertEquals(testPet, petResponse);
    }

    @Test
    public void deletePetFromStoreTest() {
        JPet testPet = new JPet(null,
                "Pet_" + RandomStringUtils.randomAlphabetic(8), //Pet name
                RandomStringUtils.randomNumeric(8), //Pet ID
                null, null,
                "available"); //Pet status

        JPetController petAction = new JPetController();
        petAction.addNewPet(testPet);
        petAction.deletePet(testPet);

        JMessage messageResponse = petAction.getPet(testPet);

        Assert.assertTrue(messageResponse.getMessage().equals("Pet not found"));
    }

}

File: kotlinRestAssuredExample/src/test/kotlin/io/swagger/petstore/PetTestsInObjectStyle.kt
package io.swagger.petstore

import io.swagger.petstore.controllers.KPetController
import io.swagger.petstore.models.KPet
import org.apache.commons.lang3.RandomStringUtils
import org.junit.Assert
import org.junit.Test

class PetTestsInObjectStyle {

    @Test
    fun `Add new pet to store test`() {
        val testPet = KPet(id = RandomStringUtils.randomNumeric(10).toLong(),
                name = "Pet_${RandomStringUtils.randomAlphabetic(8)}",
                status = "available")

        val petResponse = KPetController().addNewPet(testPet)

        Assert.assertEquals(testPet, petResponse)
    }

    @Test
    fun `Delete pet from store test`() {
        val testPet = KPet(id = RandomStringUtils.randomNumeric(10).toLong(),
                name = "Pet_${RandomStringUtils.randomAlphabetic(8)}",
                status = "available")

        KPetController().apply {
            //with instance of KPetController class call following methods:
            addNewPet(testPet)
            deletePet(testPet)

            Assert.assertTrue(getPet(testPet).message.equals("Pet not found"))
        }
    }

}